GoRogue/MapGeneration/Region.cs

```diff
@@ -1,355 +1,421 @@
 
-            Remove(connection);
-            _connections.Add(connection);
-        }
+        /// <summary>
+        /// Is this Point one of the corners of the Region?
+        /// </summary>
+        /// <param name="position">the point to evaluate</param>
+        public bool IsCorner(Point position)
+            => position == _northEastCorner || position == _northWestCorner || position == _southEastCorner || position == _southWestCorner;
 
         /// <summary>
-        /// Removes (from this region) the outer points shared in common
-        /// with the imposing region
+        /// The value of the left-most Point in the region at elevation y
         /// </summary>
-        /// <param name="imposing">the region to check for common outer points</param>
-        public void RemoveOverlappingOuterPoints(Region imposing)
-        {
-            foreach (Point c in imposing._outerPoints)
-            {
-                while (_outerPoints.Contains(c))
-                    _outerPoints.Remove(c);
-                while (_innerPoints.Contains(c))
-                    _innerPoints.Remove(c);
-            }
-        }
+        /// <param name="y">The elevation to evaluate</param>
+        /// <returns>The X-value of a Point</returns>
+        public int LeftAt(int y) => _outerPoints.LeftAt(y);
+
+        /// <summary>
+        /// The value of the right-most Point in the region at elevation y
+        /// </summary>
+        /// <param name="y">The elevation to evaluate</param>
+        /// <returns>The X-value of a Point</returns>
+        public int RightAt(int y) => _outerPoints.RightAt(y);
+
+        /// <summary>
+        /// The value of the top-most Point in the region at longitude x
+        /// </summary>
+        /// <param name="x">The longitude to evaluate</param>
+        /// <returns>The Y-value of a Point</returns>
+        public int TopAt(int x) => _outerPoints.TopAt(x);
+
+        /// <summary>
+        /// The value of the bottom-most Point in the region at longitude x
+        /// </summary>
+        /// <param name="x">The longitude to evaluate</param>
+        /// <returns>The Y-value of a Point</returns>
+        public int BottomAt(int x) => _outerPoints.BottomAt(x);
+
+        /// <inheritdoc />
+        public bool Matches(IReadOnlyArea? area) => _points.Matches(area);
+
+        /// <inheritdoc />
+        public bool Contains(IReadOnlyArea? area) => _points.Contains(area);
+
+        /// <inheritdoc />
+        public bool Contains(Point position) => _points.Contains(position);
+
+        /// <inheritdoc />
+        public bool Contains(int positionX, int positionY) => Contains((positionX, positionY));
+
+        /// <inheritdoc />
+        public bool Intersects(IReadOnlyArea area) => _points.Intersects(area);
+
+        /// <inheritdoc />
+        public IEnumerator<Point> GetEnumerator() => _points.GetEnumerator();
+
+        /// <inheritdoc />
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
```

Comment:
In case the implementations of the two overloads ever diverge:
```suggestion
        IEnumerator IEnumerable.GetEnumerator() => _points.GetEnumerator();
```